packages/smooth_app/lib/pages/onboarding/country_selector.dart

```diff
@@ -48,61 +45,120 @@ class _CountrySelectorState extends State<CountrySelector> {
-        builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
-          if (snapshot.hasError) {
-            return Text('Fatal Error: ${snapshot.error}');
-          } else if (snapshot.connectionState != ConnectionState.done) {
-            return const CircularProgressIndicator();
-          }
-          return LayoutBuilder(
-              builder: (BuildContext context, BoxConstraints constraints) {
-            final double parentWidth = constraints.maxWidth;
-            return Container(
-              padding: widget.padding,
-              child: DropdownButtonFormField<Country>(
-                value: _chosenValue,
-                decoration: widget.inputDecoration,
-                selectedItemBuilder: (BuildContext context) {
-                  return _countryList
-                      .map(
-                        (Country country) => _CountrySelectorItem(
-                          country: country,
-                          parentWidth: parentWidth,
-                        ),
-                      )
-                      .toList(growable: false);
-                },
-                items: _countryList
-                    .map<DropdownMenuItem<Country>>((Country country) {
-                  final bool isSelected = _chosenValue == country;
+  Widget build(BuildContext context) {
+    final AppLocalizations appLocalizations = AppLocalizations.of(context)!;
+    return FutureBuilder<void>(
+      future: _initFuture,
+      builder: (BuildContext context, AsyncSnapshot<void> snapshot) {
+        if (snapshot.hasError) {
+          return Text('Fatal Error: ${snapshot.error}');
+        } else if (snapshot.connectionState != ConnectionState.done) {
+          return const CircularProgressIndicator();
+        }
 
-                  return DropdownMenuItem<Country>(
-                    value: country,
-                    child: DefaultTextStyle.merge(
-                      child: _CountrySelectorItem(
-                        country: country,
-                        parentWidth: parentWidth,
-                      ),
-                      style: TextStyle(
-                        fontWeight: isSelected ? FontWeight.bold : null,
+        return GestureDetector(
+          onTap: () async {
+            List<Country> filteredList = List<Country>.from(_countryList);
+            await showDialog<Country>(
+              context: context,
+              builder: (BuildContext context) {
+                return StatefulBuilder(
+                  builder: (BuildContext context,
+                      void Function(VoidCallback fn) setState) {
+                    return AlertDialog(
+                      title: const Text('Choose your country'),
+                      // TODO(aman): translations
+                      content: SizedBox(
+                        width: 300,
+                        child: Column(
+                          children: <Widget>[
+                            TextField(
```

Comment:
I guess you could use our `SmoothTextFormField` (don't know if spelled corectly) here, then we can keep design changes continuous. But if for some reason it doesn't work out you can leave it like this